Add yaml output plugin

leedm777 opened the pull request:
##### SUMMARY
Using YAML instead of JSON for detailed output can (IMO) improve
readability; especially for tasks with either lots of output, or
multi-line output.

<!---
If you are fixing an existing issue, please include "Fixes #nnn" in your
commit message and your description; but you should still explain what
the change does.
-->

##### ISSUE TYPE
<!--- Pick one below and delete the rest: -->
 - Feature Pull Request

##### COMPONENT NAME
`yaml` callback plugin

##### ANSIBLE VERSION
<!--- Paste verbatim output from "ansible --version" between quotes below -->
```
ansible 2.5.0 (yaml-output b385c7d57a) last updated 2017/10/27 07:25:14 (GMT -500)
  config file = /Users/dlee/.ansible.cfg
  configured module search path = [u'/Users/dlee/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /Users/dlee/prj/ansible/lib/ansible
  executable location = /Users/dlee/prj/ansible/bin/ansible
  python version = 2.7.14 (default, Sep 25 2017, 09:53:22) [GCC 4.2.1 Compatible Apple LLVM 9.0.0 (clang-900.0.37)]
```


##### ADDITIONAL INFORMATION
<!---
Include additional information to help people understand the change here.
For bugs that don't have a linked bug report, a step-by-step reproduction
of the problem is helpful.
  -->

<!--- Paste verbatim command output below, e.g. before and after your change -->
```

```

leedm777 committed:
Add yaml output plugin

Using YAML instead of JSON for detailed output can (IMO) improve
readability; especially for tasks with either lots of output, or
multi-line output.

bcoca commented on lib/ansible/plugins/callback/yaml.py:
```diff
@@ -0,0 +1,118 @@
+# Make coding more python3-ish
```
missing license/copyright headers
```
# (c) 2017 Ansible Project
# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
```

leedm777 replied:
fixed

bcoca commented on CHANGELOG.md:
```diff
@@ -28,6 +28,7 @@ Ansible Changes By Release
 * Updated the bundled copy of the six library to 1.11.0
 * Added support to `become` `NT AUTHORITY\System`, `NT AUTHORITY\LocalService`, and `NT AUTHORITY\NetworkService` on Windows hosts
 * Added `aws_ssm` lookup plugin
+* Added `yaml` output plugin
```
plugins get their own section, should not be under 'minor'

bcoca commented on lib/ansible/plugins/callback/yaml.py:
```diff
@@ -0,0 +1,118 @@
+
+        if abridged_result:
+            dumped += '\n'
+            dumped += yaml.safe_dump(json.loads(json.dumps(abridged_result, ensure_ascii=False)), width=1000, default_flow_style=False)
```
there is an ansible specific dumper we create, not sure if pertinent to this plugin, but worth looking into as i think it can save you work

leedm777 replied:
Using this now

leedm777 committed:
Fix yaml output plugin documentation

 * Added copyright header to yaml.py
 * Moved mention of yaml output plugin to `New Modules` section

leedm777 committed:
Use AnsibleDumper in yaml output plugin

Also moved the `represent_scalar` setup into the init method. As a
global statement, it was causing exceptions trying to get an `id`
field that does not exist.

leedm777 committed:
Remove trailing whitespace

bcoca commented on lib/ansible/plugins/callback/yaml.py:
```diff
@@ -0,0 +1,120 @@
+
+        if abridged_result:
+            dumped += '\n'
+            dumped += yaml.dump(json.loads(json.dumps(abridged_result, ensure_ascii=False)), width=1000, Dumper=AnsibleDumper, default_flow_style=False)
```
why the json load + dump? this is effectively a noop, you can just use yaml.dump directly

leedm777 replied:
Removed

leedm777 committed:
Remove useless json.loads/dumps from yaml output
